### PR TITLE
Update_spending_plan

### PR DESCRIPTION
**Pull request by nhatnguyen71298**
Screenshot from 2021-04-06 16-09-03
Screenshot from 2021-04-06 16-09-10

- Update plan.

https://edu-redmine.sun-asterisk.vn/issues/34859?issue_count=33&issue_position=21&next_issue_id=34858&prev_issue_id=34860

### CODE REVIEW

**Comment by nhatnguyen71298** [comment]
ready

File: app/controllers/spending_plans_controller.rb [hunk 1]
```diff
@@ -18,6 +18,18 @@ def index
                                       per_page: Settings.paginate.per_page
   end
 
+  def edit; end
+
+  def update
+    @spending_plan.budget_id = @budget.id
+    if @spending_plan.update spending_plan_params
```
Comment (vanvtt-0952) [inline app/controllers/spending_plans_controller.rb:25]: chị chưa thấy chú truyền user ở đâu hết nhỉ

File: app/controllers/spending_plans_controller.rb [hunk 2]
```diff
@@ -18,6 +18,18 @@ def index
...
+      flash[:success] = t "flash.update_plan_success"
+      redirect_to @spending_plan
+    else
+      render "edit"
```
Comment (vanvtt-0952) [inline app/controllers/spending_plans_controller.rb]: `:edit` dùng symbol nhé

File: app/views/spending_plans/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<%= form_for @spending_plan do |f| %>
+  <%= render "shared/error_messages", object: @spending_plan %>
+  <div class="form-group">
+    <%= f.label :name, t(".label")[0] %>
+    <%= f.text_field :name, class:"form-control" %>
+  </div>
+  <div class="form-group">
+   <%= f.label :plan_type, t(".label")[1] %>
+   <%= f.select :plan_type, options_for_select(select_for_type_plan,
+    (@spending_plan.plan_type.present?) ? @spending_plan.plan_type : "income"),
```
Comment (vanvtt-0952) [inline app/views/spending_plans/_form.html.erb]: `(@spending_plan.plan_type.present?) ? @spending_plan.plan_type : "income")` 
-> `@spending_plan.plan_type || "income")` dc ko

File: app/views/spending_plans/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    class:"form-control"%>
+  </div>
+  <div class="form-group">
+    <%= f.label :budget_id, t(".label")[2] %>
+    <%= f.select :budget_id, options_for_select(select_budget,
+    (@spending_plan.budget.present?) ? @spending_plan.budget.id : -1), class:"form-control"%>
```
Comment (vanvtt-0952) [inline app/views/spending_plans/_form.html.erb:16]: indent

File: app/views/spending_plans/_form.html.erb (hunk 3 above)
Comment (nhatnguyen71298) [reply app/views/spending_plans/_form.html.erb]: được chị :D

**Comment by nhatnguyen71298** [comment]
ready

**Pull request merged**